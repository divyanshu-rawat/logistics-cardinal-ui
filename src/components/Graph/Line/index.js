import React from 'react';
import PropTypes from 'prop-types';
import { Line as RCLine } from 'recharts';

import { SIZES } from '../../utils/constants';

class Line extends RCLine {
  static propTypes = {
    ...RCLine.propTypes,
    /**
     * Stroke color, which would be a name of a color or an rgb()
     */
    color: PropTypes.string,
    /**
     * Stroke dashed style,
     */
    dash: PropTypes.bool,
    /**
     * Size of the line
     */
    size: PropTypes.oneOf(Object.values(SIZES)),
  };

  static defaultProps = {
    ...RCLine.defaultProps,
    dot: false,
  };

  getStrokeWidth = (size) => {
    let strokeSize;

    switch (size) {
      case SIZES.lg:
        strokeSize = 4;
        break;
      case SIZES.md:
        strokeSize = 3;
        break;
      case SIZES.sm:
        strokeSize = 2;
        break;
      default:
        strokeSize = 1;
    }

    return strokeSize;
  };

  render() {
    const { color, dash, size, ...props } = this.props;
    const additionalProps = {};

    if (dash) {
      additionalProps.strokeDasharray = '3 3';
    }

    if (color) {
      additionalProps.stroke = color;
    }

    if (size) {
      additionalProps.strokeWidth = this.getStrokeWidth(size);
    }

    return <RCLine {...props} {...additionalProps} />;
  }
}

export default Line;
