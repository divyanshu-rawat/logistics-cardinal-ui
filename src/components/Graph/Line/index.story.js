import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { boolean, color, select } from '@storybook/addon-knobs';

import withTests from '../../../utils/withTests';
import { GRAPH } from '../../../../.storybook/sections';

import ComposedChart from '../ComposedChart';
import Line from '.';
import { generateRandomShifts } from '../utils';
import { SIZES } from '../../utils/constants';

storiesOf(`${GRAPH}Line`, module)
  .addDecorator(withTests('Line/tests/index'))
  .add(
    'Simple',
    withInfo()(() => (
      <ComposedChart data={generateRandomShifts()}>
        <Line
          dataKey="shifts"
          dash={boolean('dash', true)}
          dot={boolean('dot', true)}
          color={color('color', 'chocolate')}
          size={select('size', Object.values(SIZES), SIZES.xs)}
        />
      </ComposedChart>
    )),
  );
