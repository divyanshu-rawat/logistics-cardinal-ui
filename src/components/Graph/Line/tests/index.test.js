import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Line from '..';

const renderedComponent = () =>
  shallow(<Line dataKey="shifts" color="chocolate" size="xs" dash />);

describe('<Line />', () => {
  it('should render the component', () => {
    const wrapper = renderedComponent();
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('should render correct sizes', () => {
    const wrapper = renderedComponent();
    const instance = wrapper.instance();
    expect(instance.getStrokeWidth('lg')).toBe(4);
    expect(instance.getStrokeWidth('md')).toBe(3);
    expect(instance.getStrokeWidth('sm')).toBe(2);
    expect(instance.getStrokeWidth('xs')).toBe(1);
  });
});
