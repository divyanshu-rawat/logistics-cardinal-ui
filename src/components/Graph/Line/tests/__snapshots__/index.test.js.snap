// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Line /> should render the component 1`] = `
<Line
  activeDot={true}
  animationBegin={0}
  animationDuration={1500}
  animationEasing="ease"
  connectNulls={false}
  dataKey="shifts"
  dot={false}
  fill="#fff"
  hide={false}
  isAnimationActive={true}
  legendType="line"
  onAnimationEnd={[Function]}
  onAnimationStart={[Function]}
  points={Array []}
  stroke="chocolate"
  strokeDasharray="3 3"
  strokeWidth={1}
  xAxisId={0}
  yAxisId={0}
/>
`;
